core/src/main/java/org/mule/runtime/core/internal/execution/SourceResultAdapter.java

```diff
@@ -31,6 +33,7 @@ public class SourceResultAdapter {
   private final MediaType mediaType;
   private final Optional<String> correlationId;
   private final PayloadMediaTypeResolver payloadMediaTypeResolver;
+  private final Optional<Object> notification;
```

Comment:
Bad impl:

* Imposes artificial limitation of a single notification per instance
* Sources already have mechanism for firing notifications
* Leads to blind cast called out above